### DISCUSSION TITLE
Can I use "Update branch protection" API to create new protection for some branchs apply branch name pattern

### DISCUSSION THREAD

**Discussion by k19810703** [community/community#24338] [Apps, API and Webhooks]
<p>Can I use branch name patern in branch feild? Description says it can be branch+ parameter , is there any examples?</p>
<h4><a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#update-branch-protection--parameters" rel="noopener nofollow ugc">Parameters</a></h4>

**Comment by parithiban** [comment, marked as answer]
<p><a class="mention" href="/k19810703">@k19810703</a> Welcome to GitHub community :wave: :wave:</p>
<p>I am not sure you can achieve it using REST V3 API but still it can be done using GraphQL V4.</p>
<p>See below for Query</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/0/b/0b12a4a9655eae87911cbce271c481ae61ef70c2.png" data-download-href="/uploads/short-url/1zXc5Q5tQ8xB39lbxdU7yucHWpA.png?dl=1" title="Screen-Shot-2020-12-27-at-11-48-23-PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/7281166/181084285-38ff29c1-f131-4a36-9cf6-e99320f41849.png" alt="Screen-Shot-2020-12-27-at-11-48-23-PM" data-base62-sha1="1zXc5Q5tQ8xB39lbxdU7yucHWpA" width="690" height="395" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/0/b/0b12a4a9655eae87911cbce271c481ae61ef70c2_2_690x395.png, https://aws1.discourse-cdn.com/github/optimized/3X/0/b/0b12a4a9655eae87911cbce271c481ae61ef70c2_2_1035x592.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/0/b/0b12a4a9655eae87911cbce271c481ae61ef70c2_2_1380x790.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/b/0b12a4a9655eae87911cbce271c481ae61ef70c2_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen-Shot-2020-12-27-at-11-48-23-PM</span><span class="informations">2868×1644 379 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p><strong>Output</strong></p>
<p>Before</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/b/cbfdf1838d3d665b1ef5371471124467cf9bad71.png" data-download-href="/uploads/short-url/t6AZRn4xj9k3jlOoUlf0rCpiHEl.png?dl=1" title="Screen Shot 2020-12-27 at 11.54.19 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/7281166/181084286-fde81a98-fc13-4949-bf77-bb2f133e3348.png" alt="Screen Shot 2020-12-27 at 11.54.19 PM" data-base62-sha1="t6AZRn4xj9k3jlOoUlf0rCpiHEl" width="685" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/b/cbfdf1838d3d665b1ef5371471124467cf9bad71_2_685x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/b/cbfdf1838d3d665b1ef5371471124467cf9bad71_2_1027x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/c/b/cbfdf1838d3d665b1ef5371471124467cf9bad71_2_1370x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/b/cbfdf1838d3d665b1ef5371471124467cf9bad71_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-12-27 at 11.54.19 PM</span><span class="informations">1968×1436 305 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>After</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/f/e/fefcab289edc43607ada5f9ef795b63d73f751ba.png" data-download-href="/uploads/short-url/AnIChtfAtx7fbma2VAFQqIm5ucy.png?dl=1" title="Screen Shot 2020-12-27 at 11.54.33 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/7281166/181084287-f2721e51-1f64-4399-b7bc-272ff9327e90.png" alt="Screen Shot 2020-12-27 at 11.54.33 PM" data-base62-sha1="AnIChtfAtx7fbma2VAFQqIm5ucy" width="689" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/f/e/fefcab289edc43607ada5f9ef795b63d73f751ba_2_689x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/f/e/fefcab289edc43607ada5f9ef795b63d73f751ba_2_1033x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/f/e/fefcab289edc43607ada5f9ef795b63d73f751ba_2_1378x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/f/e/fefcab289edc43607ada5f9ef795b63d73f751ba_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-12-27 at 11.54.33 PM</span><span class="informations">2074×1504 339 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
